pydantic/_internal/_generate_schema.py

```diff
@@ -1375,13 +1376,13 @@ def _union_schema(self, union_type: Any) -> core_schema.CoreSchema:
 
     def _type_alias_type_schema(
         self,
-        obj: TypeAliasType,
+        obj: TypeAliasType | GenericAlias,
```

Comment:
`GenericAlias` and `typing._GenericAlias` act as proxy objects in some way, I don't think we want to annotate `obj` like so here.

Reply:
Right, I forgot about that.